File: benchmark/objectivec/task7_error.m
#import <Foundation/Foundation.h>

NSDictionary* parseInt(NSString* s) {
    NSScanner* scanner = [NSScanner scannerWithString:s];
    int value;
    if ([scanner scanInt:&value] && [scanner isAtEnd]) {
        return @{@"success": @YES, @"value": @(value)};
    }
    return @{@"success": @NO, @"error": @"Invalid integer format"};
}

File: benchmark/delphi/task7_error.pas
type
  TParseResult = record
    Success: Boolean;
    Value: Integer;
    Error: string;
  end;

function ParseInt(const S: string): TParseResult;
var
  Code: Integer;
begin
  Val(S, Result.Value, Code);
  Result.Success := Code = 0;
  if not Result.Success then
    Result.Error := 'Invalid integer format';
end;
